openhands/server/github.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+import httpx
+
+from openhands.core.logger import openhands_logger as logger
+
+GITHUB_CLIENT_ID = os.getenv('GITHUB_CLIENT_ID', '').strip()
+GITHUB_CLIENT_SECRET = os.getenv('GITHUB_CLIENT_SECRET', '').strip()
+GITHUB_USER_LIST = None
+
+
+def load_github_user_list():
+    global GITHUB_USER_LIST
+    waitlist = os.getenv('GITHUB_USER_LIST_FILE')
+    if waitlist:
+        with open(waitlist, 'r') as f:
+            GITHUB_USER_LIST = [line.strip() for line in f if line.strip()]
+
+
+load_github_user_list()
```

Comment:
The server will require a restart every time a new user is added. We could instead read on every call to `authenticate_github_user` and only update the txt file without requiring a restart

Reply:
We'll never update the list without restarting the server in tandem, because kubernetes 🙃